src/couch/src/couch_db.erl

```diff
@@ -297,7 +303,13 @@ open_doc(Db, IdOrDocInfo) ->
     open_doc(Db, IdOrDocInfo, []).
 
 open_doc(Db, Id, Options) ->
-    increment_stat(Db, [couchdb, database_reads]),
+    open_doc(Db, Id, Options, true).
+
+open_doc(Db, Id, Options, ChangeStat) ->
+    case ChangeStat of
+        true -> ?MODULE:increment_stat(Db, [couchdb, database_reads]);
+        false -> ok
+    end,
```

Comment:
I'd put this setting inside the Options list, however shouldn't database_reads be incremented whenever we read? open_doc_int/3 will open the doc, so why wouldn't we count it? It seems the path in question opens the same doc twice? If so, that's not a double counting error in my view, incrementing twice seems correct?

Reply:
Yeah we should be triggering database reads anytime a read is performed. The bug here is that we read the doc twice, one for include_docs=true and once for the filter, when we should only be doing that once.

It's a bit more complicated in that the js filter will potentially load all the available revs, whereas include_docs does not, so some care will need to be taken there, but we definitely do not want to hide data from stats collection.

Reply:
Updated the PR to only read docs once when filtering `_change` is triggered.
May I get another review? Thanks.